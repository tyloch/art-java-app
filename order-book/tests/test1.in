{"type": "Limit", "order": {"direction": "Buy", "id":1, "price": 14, "quantity":20}}
{"type": "Iceberg", "order": {"direction": "Buy", "id":2, "price": 15, "quantity":50, "peak": 20}}
{"type": "Limit", "order": {"direction": "Sell", "id":3, "price": 16, "quantity": 15}}
{"type": "Limit", "order": {"direction": "Sell", "id":4, "price": 13, "quantity": 60}}