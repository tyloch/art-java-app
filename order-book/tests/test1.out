{"buyOrders": [{"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrders": [{"id": 2, "price": 15, "quantity": 20}, {"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrders": [{"id": 2, "price": 15, "quantity": 20}, {"id": 1, "price": 14, "quantity": 20}], "sellOrders": [{"id": 3, "price": 16, "quantity": 15}]}
{"buyOrders": [{"id": 1, "price": 14, "quantity": 10}], "sellOrders": [{"id": 3, "price": 16, "quantity": 15}]}
{"buyOrderId": 2, "sellOrderId": 4, "price": 15, "quantity": 20}
{"buyOrderId": 2, "sellOrderId": 4, "price": 15, "quantity": 20}
{"buyOrderId": 2, "sellOrderId": 4, "price": 15, "quantity": 10}
{"buyOrderId": 1, "sellOrderId": 4, "price": 14, "quantity": 10}
